### PR TITLE
Desktop: Fixed order of notebooks in moveToNotebook dialog

### LINKED ISSUE #3052: "move to notebook" list not sorted in alphabetic order
## Environment

Joplin version: 1.0.200
Platform: linux
OS specifics: debian

## Steps to reproduce

1.  when moving note to another notebook the popup list is not sorted in alpha order

### PR DESCRIPTION
**Pull request by coderrsid**
fixes #3052 
I tried to test all kind of edge cases. 

### Before 
Screen-Recording-2020-04-18-at-4

### After
Screen-Recording-2020-04-18-at-4 (1)

### CODE REVIEW

**Comment by coderrsid** [comment]
Yeah laurent, for sure. That's what i also asked in the issue. Thanks for telling i'm working on this 👍😄

File: ElectronClient/gui/MainScreen.jsx [hunk 1]
```diff
@@ -242,17 +242,38 @@ class MainScreenComponent extends React.Component {
...
+			let startFolders = [];
 			const maxDepth = 15;
 
+			const folderSort = (folders) => {
+				folders.sort((a, b) => {
+					if (a.indentDepth && b.indentDepth) {
+						if (a.indentDepth === b.indentDepth && a.parentId === b.parentId) {
+							// sensitivity accent will treat accented characters as different
+							// but treats caps as equal
+							return a.label.localeCompare(b.label, undefined, { sensitivity: 'accent' });
```
[the remark this answers is not in the document]
Comment (coderrsid) [reply ElectronClient/gui/MainScreen.jsx:255]: Thanks @Rishgod. Although in open source projects, compatibility is a bigger problem and as laurent suggested to add the `sortFolderTree` function to Folder model, so it will be for sure used for future cases too and if you see compatibility of `Intl.Collator`, it's not at all good with Android Webview. 
image

But for `Sort`, it is quite compatible. So, i'll rather suggest myself to go with this as it will not be a performance issue as much as i know.
image

**Comment by coderrsid** [comment]
Hey, @tessus  or @laurent22 . Can you open

File: ElectronClient/gui/MainScreen.jsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (Rishgod) [reply ElectronClient/gui/MainScreen.jsx]: Umm, compatibility should be compared with localCompare not sort because init collator will replace that.

[the remark this answers is not in the document]
Comment (coderrsid) [reply ElectronClient/gui/MainScreen.jsx]: Yeah, sorry it just got mixed up. But yeah the compatibility of localeCompare is more. Although thanks for putting out options.

**Comment by coderrsid** [comment]
@PackElend  label me please ! Thanks 😄

**Comment by coderrsid** [comment]
@laurent22  I've added the `Folder::sortFolderTree` and `Unit-tests` related to it as you said. Please review it as soon as you get time. 😄

**Comment by laurent22** [comment]
That looks good, thanks @coderrsid
